diff --git a/mindspeed/core/fusions/fused_layer_norm.py b/mindspeed/core/fusions/fused_layer_norm.py
index 89ea8614..3e047b22 100644
--- a/mindspeed/core/fusions/fused_layer_norm.py
+++ b/mindspeed/core/fusions/fused_layer_norm.py
@@ -7,7 +7,7 @@ import torch
 class FusedLayerNormAffineFunction:
     @staticmethod
     def apply(input_, weight, bias, normalized_shape, eps):
-        return torch.nn.functional.layer_norm(input_, normalized_shape, weight, bias, eps)
+        return torch.nn.functional.layer_norm(input_.float(), normalized_shape, weight.float(), bias.float(), eps).type_as(input_)
 
     @staticmethod
     def forward(*args, **kwargs):
diff --git a/mindspeed/core/fusions/fused_rope.py b/mindspeed/core/fusions/fused_rope.py
index 267fad59..1774dd92 100644
--- a/mindspeed/core/fusions/fused_rope.py
+++ b/mindspeed/core/fusions/fused_rope.py
@@ -61,6 +61,7 @@ def apply_rotary_pos_emb_bshd(
         t = torch.cat((x1, x2), dim=-1)
 
     rot_dim = freqs.shape[-1]
+    t_type = t.dtype
     t, t_pass = t[..., :rot_dim], t[..., rot_dim:]
     cos_ = (torch.cos(freqs) * _mscale).to(t.dtype)
     sin_ = (torch.sin(freqs) * _mscale).to(t.dtype)
@@ -69,7 +70,9 @@ def apply_rotary_pos_emb_bshd(
         mode = 1 if rotary_interleaved else 0
         t = npu_rotary_position_embedding(t.contiguous(), cos_, sin_, mode).to(t.dtype)
     else:
-        t = (t * cos_) + (_rotate_half(t, rotary_interleaved) * sin_)
+        # t = (t * cos_) + (_rotate_half(t, rotary_interleaved) * sin_)
+        t = (t.float() * cos_.float()) + (_rotate_half(t.float(), rotary_interleaved) * sin_.float())
+        t = t.to(t_type)
 
     return torch.cat((t, t_pass), dim=-1)
 
diff --git a/mindspeed/core/transformer/experimental_attention_variant/dsa_fused.py b/mindspeed/core/transformer/experimental_attention_variant/dsa_fused.py
index 547e938b..de145d3f 100644
--- a/mindspeed/core/transformer/experimental_attention_variant/dsa_fused.py
+++ b/mindspeed/core/transformer/experimental_attention_variant/dsa_fused.py
@@ -276,8 +276,8 @@ def forward_with_scores(
     # =========================================
     # Rotate activation
     # =========================================
-    q = rotate_activation(q)
-    k = rotate_activation(k)
+    # q = rotate_activation(q)
+    # k = rotate_activation(k)
 
     # =========================================
     # Compute index scores

